modules/dtj/Animal/scripts/barnyard.pl

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+
+use FindBin;
+use lib "$FindBin::Bin/../lib";
+
+use Cow;
+use Horse;
+use Sheep;
+use Mouse;
+
+my @animals;
+print "Enter some Animals to go into the barnyard, one per line\n";
+print "Press ctrl+d when finished\n";
+while ( <> ) {
+    chomp;
+    my $input = $_;
+
+    push @animals, $input;
+}
+
+foreach my $animal ( @animals ) {
+    eval {
+        $animal->speak;
```

Comment:
OK, what deep voodoo makes this work? I thought `$animal` would be a string at this point.